dali/base/dadfs.cpp

```diff
@@ -585,6 +585,37 @@ class CFileSuperOwnerLock : protected CFileLockCompound
     {
         return CFileLockCompound::detach();
     }
+    bool initWithFileLock(const CDfsLogicalFileName &logicalName, IRemoteConnection *conn, unsigned timeout, const char *msg, CFileLock fcl, unsigned fclmode)
+    {
+        IRemoteConnection *fclConn = fcl.queryConnection();
+        CTimeMon tm(timeout);
+        unsigned interval = 5000;
+        unsigned remaining = timeout;
+        loop {
+            verifyex(!tm.timedout(&remaining));
+            if (interval > remaining)
+                interval = remaining;
+            try
+            {
+                if (init(logicalName, conn, interval, msg))
+                    return true;
+            }
+            catch (IException *e)
+            {
+                // should really continue only when timeout occurs
```

Comment:
Yes I agree..
should change to abort/rethrow error if not a SDSExcpt_LockTimeout exception.